### PR TITLE
Update template with mui 5 and emotion

### PR DESCRIPTION
**Pull request by MaxJGruber**

### CODE REVIEW

**Comment by MaxJGruber** [comment]
@finnwoelm-sdsn , everything works! But for the life of me, I do not understand why the SampleDrawer component is not rendering...

File: drawers/AppDrawer.js [hunk 1]
```diff
@@ -98,7 +52,7 @@ const AppDrawer = observer(({ children, permanent = false }) => {
     return () => {
       disposeReaction();
     };
-  }, []);
+  }, [uiStore.target]);
```
Comment (finnwoelm-sdsn) [inline drawers/AppDrawer.js:56]: This comes from the linting?

File: drawers/SampleDrawer.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
-import { Typography } from "@material-ui/core";
+import { Typography } from "@mui/material";
 
 const SampleDrawer = () => (
+  // eslint-disable-next-line
```
Comment (finnwoelm-sdsn) [inline drawers/SampleDrawer.js:4]: Why is linting unhappy?

File: README.md [hunk 3]
```diff
@@ -37,9 +37,9 @@ const MyButton = styled(button)`
 
 We use [Next SEO](https://github.com/garmeeh/next-seo) for search engine optimization. The default SEO is configured in `config.js` and set up in `pages/_app.js`. The defaults can be overwritten on a per-page basis. See: https://github.com/garmeeh/next-seo#default-seo-configuration
 
-## Notes
+### Linting
 
-### Fast Refresh
+Linting has been set up. Run `npm run lint` to lint all files.
 
 On Windows, [Fast Refresh](https://nextjs.org/docs/basic-features/fast-refresh) (hot reloading) only works with Google Chrome.
```
Comment (finnwoelm-sdsn) [inline README.md:45]: Broken merge.

File: components/Button.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        p: 2,
+        minWidth: 300,
+      }}
+    >
+      works
+    </Button>
+  </Stack>
+);
+
+export default BasicButtons;
```
Comment (finnwoelm-sdsn) [inline components/Button.js:27]: Can remove

File: components/SampleToggleDrawerButton.js [hunk 5]
```diff
@@ -1,13 +1,12 @@
...
-import { Button } from "@material-ui/core";
+import { Button } from "@mui/material";
 import { useUiStore } from "stores/uiStore";
 
 const SampleToggleDrawerButton = observer(() => {
   const uiStore = useUiStore();
 
   return (
     <Button
-      color="primary"
```
Comment (finnwoelm-sdsn) [inline components/SampleToggleDrawerButton.js:10]: Nice catch 👍

File: drawers/AppDrawer.js [hunk 6]
```diff
@@ -64,7 +15,11 @@ const scrollElementToTop = (elementId) => {
 };
 
 const AppDrawer = observer(({ children, permanent = false }) => {
+  console.log(children);
```
Comment (finnwoelm-sdsn) [inline drawers/AppDrawer.js:18]:
```suggestion
```

File: pages/_app.js [hunk 7]
```diff
@@ -10,20 +9,12 @@ import { TITLE, META_DESCRIPTION, META_IMAGE, URL } from "root/config";
 import ThemeProvider from "components/ThemeProvider";
 import * as gtag from "helpers/gtag";
 
-function MyApp({ Component: NextPage, pageProps }) {
+const App = ({ Component: NextPage, pageProps }) => {
```
Comment (finnwoelm-sdsn) [inline pages/_app.js:12]: function to const comes from lint? Just wondering

**Review by finnwoelm-sdsn — COMMENTED** [review]
Any leads on the SampleDrawer issue?

File: drawers/AppDrawer.js (hunk 1 above)
Comment (MaxJGruber) [reply drawers/AppDrawer.js:56]: Yes

File: drawers/SampleDrawer.js (hunk 2 above)
Comment (MaxJGruber) [reply drawers/SampleDrawer.js:4]: Doesn't like the apostrophe in " I'm " apparently.

Comment (finnwoelm-sdsn) [reply drawers/SampleDrawer.js:4]: What? 🤣🤣 my gosh

File: pages/_app.js (hunk 7 above)
Comment (MaxJGruber) [reply pages/_app.js]: Nope I guess I just had it as const in the merge

Comment (finnwoelm-sdsn) [reply pages/_app.js]: Ok, that's fine! Was just curious :)

File: drawers/SampleDrawer.js (hunk 2 above)
Comment (finnwoelm-sdsn) [reply drawers/SampleDrawer.js:4]: Ok, we should fix this with &apos; as is suggested: https://stackoverflow.com/a/45898131

There is a good reason for this detailed here: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md#prevent-invalid-characters-from-appearing-in-markup-reactno-unescaped-entities

**Comment by finnwoelm-sdsn** [comment]
@MaxJGruber on the drawer not rendering, that's my bad. I think it needs to be like this:

```js
Index.layoutProps = () => ({ Drawer: <SampleDrawer /> });
```

(with opening and closing tags)

Can you double check that this is consistent with what we have in the global dashboard?

**Comment by MaxJGruber** [comment]
@finnwoelm-sdsn Yes! That's what it was!

File: README.md [hunk 8]
```diff
@@ -54,3 +48,5 @@ git remote add nextjs-starter https://github.com/sdsna/nextjs-starter.git
 git fetch nextjs-starter
 git merge nextjs-starter/main
 ```
+
+Linting (and code formatting) also occurs automatically as a pre-commit hook thanks to [lint-staged](https://github.com/okonet/lint-staged) and [husky](https://github.com/typicode/husky).
```
Comment (finnwoelm-sdsn) [inline README.md:52]: This is still broken compared to nextjs-starter: Missing `Linting` header plus first line. (not a major deal, but better to fix)

**Pull request merged**